baml_language/crates/baml_thir/src/exhaustiveness.rs

```diff
@@ -0,0 +1,833 @@
+                }
+                values
+            }
+
+            // Named type: could be enum, class, or type alias
+            Ty::Named(name) => {
+                // Check if it's a type alias
+                if let Some(alias_ty) = self.type_aliases.get(name) {
+                    return self.expand_type_to_values(alias_ty);
+                }
+
+                // Check if it's an enum (finite type)
+                if let Some(variants) = self.enum_variants.get(name) {
+                    return variants
+                        .iter()
+                        .map(|variant_name| ValueSet::EnumVariant {
+                            enum_name: name.clone(),
+                            variant_name: variant_name.clone(),
+                        })
+                        .collect();
+                }
+
+                // Unknown named type or class (infinite)
+                vec![ValueSet::OfType(name.clone())]
+            }
+
+            // Bool is finite: {true, false}
+            Ty::Bool => vec![
+                ValueSet::Literal(Literal::Bool(true)),
+                ValueSet::Literal(Literal::Bool(false)),
+            ],
+
+            // Null is a singleton
+            Ty::Null => vec![ValueSet::Literal(Literal::Null)],
+
+            // Singleton types (literal types like `200` in `200 | 201`)
+            Ty::Singleton(sv) => {
+                use crate::SingletonValue;
+                match sv {
+                    SingletonValue::Int(v) => vec![ValueSet::Literal(Literal::Int(*v))],
+                    SingletonValue::String(v) => {
+                        vec![ValueSet::Literal(Literal::String(v.clone()))]
+                    }
+                    SingletonValue::Bool(v) => vec![ValueSet::Literal(Literal::Bool(*v))],
+                    SingletonValue::EnumVariant(enum_name, variant_name) => {
+                        vec![ValueSet::EnumVariant {
+                            enum_name: enum_name.clone(),
+                            variant_name: variant_name.clone(),
+                        }]
+                    }
+                }
+            }
+
+            // Infinite types: int, float, string, classes, etc.
+            Ty::Int => vec![ValueSet::OfType(Name::new("int"))],
+            Ty::Float => vec![ValueSet::OfType(Name::new("float"))],
+            Ty::String => vec![ValueSet::OfType(Name::new("string"))],
+            Ty::Image => vec![ValueSet::OfType(Name::new("image"))],
+            Ty::Audio => vec![ValueSet::OfType(Name::new("audio"))],
+            Ty::Video => vec![ValueSet::OfType(Name::new("video"))],
+            Ty::Pdf => vec![ValueSet::OfType(Name::new("pdf"))],
+            Ty::Class(_) => vec![ValueSet::OfType(Name::new("<class>"))],
+            Ty::Enum(_) => vec![ValueSet::OfType(Name::new("<enum>"))],
+            Ty::List(_) => vec![ValueSet::OfType(Name::new("<list>"))],
+            Ty::Map { .. } => vec![ValueSet::OfType(Name::new("<map>"))],
```

Comment:
Does this not distinguish between maps of different types?
- `map<string, int>` 
- `map<string, string>`
- `map<Status, string>`

In our legacy code map keys can be strings or enums. Any others were not supported. That would be it's own PR but doesn't the code here need to check the types of key and value?